api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -3977,4 +3982,99 @@ public void saveOrder_shouldSaveTheFormNamespaceAndPath() {
+		OrderAttributeType orderAttributeType = orderService.getOrderAttributeTypeByUuid(
+				"9758d106-79b0-4f45-8d8c-ae8b3f25d72a");
+		assertEquals("Referral", orderAttributeType.getName());
+	}
+
+	@Test
+	public void saveOrderAttributeType_shouldSaveTheProvidedOrderAttributeType()
+			throws ParseException {
+		OrderAttributeType orderAttributeType = new OrderAttributeType();
+		orderAttributeType.setName("Medical Procedures");
+		orderAttributeType.setDatatypeClassname(FreeTextDatatype.class.getName());
+		orderService.saveOrderAttributeType(orderAttributeType);
+		assertNotNull(orderAttributeType.getId());
+	}
+
+	@Test
+	public void saveOrderAttributeType_shouldEditTheExistingOrderAttributeType() {
+		OrderAttributeType orderAttributeType = orderService.getOrderAttributeTypeById(4);
+		assertEquals("Drug", orderAttributeType.getName());
+		orderAttributeType.setName("Drug Dispense");
+		orderService.saveOrderAttributeType(orderAttributeType);
+		assertEquals("Drug Dispense", orderAttributeType.getName());
+	}
+
+	@Test
+	public void retireOrderAttributeType_shouldRetireTheProvidedOrderAttributeType() throws ParseException {
+		OrderAttributeType orderAttributeType = orderService.getOrderAttributeTypeById(2);
+		assertFalse(orderAttributeType.getRetired());
+		assertNull(orderAttributeType.getRetiredBy());
+		assertNull(orderAttributeType.getRetireReason());
+		assertNull(orderAttributeType.getDateRetired());
+		orderService.retireOrderAttributeType(orderAttributeType, "Test Retire");
+		orderAttributeType = orderService.getOrderAttributeTypeById(2);
+		assertTrue(orderAttributeType.getRetired());
+		assertNotNull(orderAttributeType.getRetiredBy());
+		assertEquals("Test Retire", orderAttributeType.getRetireReason());
+		assertNotNull(orderAttributeType.getDateRetired());
+	}
+
+	@Test
+	public void unretireOrderAttributeType_shouldUnretireTheProvidedOrderAttributeType() {
+		OrderAttributeType orderAttributeType = orderService.getOrderAttributeTypeById(4);
+		assertTrue(orderAttributeType.getRetired());
+		assertNotNull(orderAttributeType.getRetiredBy());
+		assertNotNull(orderAttributeType.getDateRetired());
+		assertNotNull(orderAttributeType.getRetireReason());
+		orderService.unretireOrderAttributeType(orderAttributeType);
+		assertFalse(orderAttributeType.getRetired());
+		assertNull(orderAttributeType.getRetiredBy());
+		assertNull(orderAttributeType.getDateRetired());
+		assertNull(orderAttributeType.getRetireReason());
+	}
+
+	@Test
+	public void getOrderAttributeTypeByName_shouldReturnOrderAttributeTypeUsingProvidedName() {
+		OrderAttributeType orderAttributeType = orderService.getOrderAttributeTypeByName("Referral");
+		assertEquals("9758d106-79b0-4f45-8d8c-ae8b3f25d72a", orderAttributeType.getUuid());
+	}
+
+	@Test
+	public void purgeOrderAttributeType_shouldPurgeTheProvidedOrderAttributeType() {
+		orderService.purgeOrderAttributeType(orderService.getOrderAttributeTypeById(3));
```

Comment:
It is necessary to show the previous size of the list before purging  e.g
`assertEquals(4, orderService.getAllOrderAttributeTypes().size());`